flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/KvStateService.java

```diff
@@ -170,7 +170,7 @@ public static KvStateService fromConfiguration(TaskManagerServicesConfiguration
 			int numProxyServerQueryThreads = qsConfig.numProxyQueryThreads() == 0 ?
 				taskManagerServicesConfiguration.getNumberOfSlots() : qsConfig.numProxyQueryThreads();
 			kvClientProxy = QueryableStateUtils.createKvStateClientProxy(
-				taskManagerServicesConfiguration.getTaskManagerAddress(),
+				taskManagerServicesConfiguration.getExternalAddress(),
```

Comment:
Not sure, whether this change is valid. The problem is that the underlying Netty server uses this address to bind against. If the external address is not resolvable, it might fail.

Reply:
As mentioned in the JIRA ticket, I'd like to exclude queryable state from the scope of this PR. Because I'm not familiar with how it works, and it seems does not have many use cases.

I think this should not affect other use cases, because the code path should only be executed if queryable state is enabled.

I think we should mention in the release notes that the new NAT support does not cover queryable state.

Reply:
Sounds good.